config/routes.rb

```diff
@@ -29,6 +29,8 @@
 
   get "/users", to: redirect("/users/active")
   get "/users/:user_state" => "users#index", :constraints => {user_state: /(in)?active/}, :as => "users_index"
+  get "/users/:id/deactivate" => "users#deactivate", :as => "deactivate_user"
+  get "/users/:id/reactivate" => "users#reactivate", :as => "reactivate_user"
   resources :users, except: [:index]
```

Comment:
You might get away with slightly less of the `:as` and `=>` here:

```suggestion
  resources :users, except: [:index] do
    member do
      get :deactivate
      get :reactivate
    end
  end
```

You might *also* be able to do something with `/users/:user_state` to simplify, but that existed before this PR so 😴 🐶 and all that...